### PR TITLE
feat: oneof=unions-value to use the same field name for oneof cases

### LINKED ISSUE #1060: Easier to use oneof=union generated code
Today, if I enable `oneof=union`, my union might generate code like this:

```ts
type?:
  | { $case: "float"; float: FloatOptions }
  | { $case: "double"; double: DoubleOptions }
  | { $case: "int32"; int32: Int32Options }
```

This can be really difficult to deal with in code, because the name of the property depends on the value of `$case`. For example:

```
switch (v.$case) {
  case "float":
  case "double":
    // I can't write generic code to handle both, because one is accessed with v.float, while the other is v.double!
	break;
  case "int32":
    console.log(v.int32.whatever);
    break;
}
```

This also inspired a lot of the complex type helpers I use (see #1034).

A nicer codegen would be:

```ts
type?:
  | { $case: "float"; value: FloatOptions }
  | { $case: "double"; value: DoubleOptions }
  | { $case: "int32"; value: Int32Options }
```

Which then allows:

```
switch (v.$case) {
  case "float":
  case "double":
    console.log(v.value.whateverFloat);
	break;
  case "int32":
    console.log(v.value.whateverInt);
    break;
}
```

This would have to be a new option to avoid breaking previously generated code.

### PR DESCRIPTION
**Pull request by bhollis**
Fixes #1060.

This adds a new `oneof=unions-value` option that changes how union oneof code is generated. Instead of each case having a field named the same as the `$case`, each of them has the same field, called `value`. This should simplify writing generic code that can handle multiple cases at once.

I chose to make it another option for `oneof=` instead of a separate option since it's just another way of handling oneofs. I also updated the README but it may be a bit much, I'm happy to remove some stuff. I also presumptively suggest that this be the new recommended oneof option.

### CODE REVIEW

File: README.markdown [hunk 1]
```diff
@@ -752,7 +752,7 @@ ts-protoc --ts_proto_out=./output -I=./protos ./protoc/*.proto
 # Todo
 
 - Support the string-based encoding of duration in `fromJSON`/`toJSON`
-- Make `oneof=unions` the default behavior in 2.0
+- Make `oneof=unions-value` the default behavior in 2.0
```
Comment (stephenh) [inline README.markdown:755]: Ah nice! I definitely agree this `value` field is better and should be the default.

We're in luck that a 2.x release is likely going to happen in the next ~month or so, via #1058, so we should plan on flipping this default over, and honestly probably just removing the option/old behavior all together.

**Review by stephenh — APPROVED** [review]

**Comment by stephenh** [comment]
This is great, thank you @bhollis !

File: README.markdown (hunk 1 above)
Comment (bhollis) [reply README.markdown:755]: Sounds great, feel free to ping me if you want help with that.

Comment (stephenh) [reply README.markdown:755]: Great! I'll ping you once Timo's PR lands as an 2.0.0 alpha and would be great to get this flipped over.

**Pull request merged**